### PR TITLE
Toolkit: add junit reporting and jest.config.js to plugin build

### PR DESCRIPTION
**Pull request by ryantxu**
When running tests in vscode, `jest` is called directly -- so for testing to work consistently, we need the same jest.config.js running for `yarn test` and plain `jest`

This PR:
* adds a dummy `jest.config.js` that loads a common file from toolkit
* includes the JUnit test report

### CODE REVIEW

File: packages/grafana-toolkit/src/config/jest.plugin.config.ts [hunk 1]
```diff
@@ -80,3 +94,16 @@ export const jestConfig = (baseDir: string = process.cwd()) => {
     ...otherOverrides,
   };
 };
+
+/**
+ * This will load the existing just setup, or use the default if it exists
+ */
+export const loadJestPluginConfig = (baseDir: string = process.cwd()) => {
+  const cfgpath = path.resolve(baseDir, 'jest.config.js');
+  if (!fs.existsSync(cfgpath)) {
```
Comment (stevenvachon) [inline packages/grafana-toolkit/src/config/jest.plugin.config.ts]: It's a best practice to always use async functions when possible. Any defined function can then be used concurrently.

File: packages/grafana-toolkit/src/config/jest.plugin.config.ts [hunk 2]
```diff
@@ -80,3 +94,16 @@ export const jestConfig = (baseDir: string = process.cwd()) => {
...
+    const src = path.resolve(baseDir, 'node_modules/@grafana/toolkit/src/config/jest.plugin.config.local.js');
+    fs.copyFileSync(src, cfgpath);
```
Comment (stevenvachon) [inline packages/grafana-toolkit/src/config/jest.plugin.config.ts]: async

File: packages/grafana-toolkit/src/config/jest.plugin.config.ts (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-toolkit/src/config/jest.plugin.config.ts]: I am nervous about requiring the whole call stack to become async... all to support somehting that is identical and should not be called async

Comment (stevenvachon) [reply packages/grafana-toolkit/src/config/jest.plugin.config.ts]: Okay, can be fixed later

**Review by stevenvachon — APPROVED** [review]

**Pull request merged**